src/AzureRMTools.ts

```diff
@@ -321,6 +286,145 @@ export class AzureRMTools {
+        };
+        deploymentTemplateFileSubscriptionsArray.push(vscode.languages.registerCompletionItemProvider(supportedDocumentSelector, completionProvider, "'", "[", "."));
+
+        const definitionProvider: vscode.DefinitionProvider = {
+            provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): vscode.Definition {
+                return self.onProvideDefinition(document, position, token);
+            }
+        };
+        deploymentTemplateFileSubscriptionsArray.push(vscode.languages.registerDefinitionProvider(supportedDocumentSelector, definitionProvider));
+
+        const referenceProvider: vscode.ReferenceProvider = {
+            provideReferences(document: vscode.TextDocument, position: vscode.Position, context: vscode.ReferenceContext, token: vscode.CancellationToken): vscode.Location[] {
+                return self.onProvideReferences(document, position, context, token);
+            }
+        };
+        deploymentTemplateFileSubscriptionsArray.push(vscode.languages.registerReferenceProvider(supportedDocumentSelector, referenceProvider));
+
+        const signatureHelpProvider: vscode.SignatureHelpProvider = {
+            provideSignatureHelp(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.SignatureHelp> {
+                return self.onProvideSignatureHelp(document, position, token);
+            }
+        };
+        deploymentTemplateFileSubscriptionsArray.push(vscode.languages.registerSignatureHelpProvider(supportedDocumentSelector, signatureHelpProvider, ",", "(", "\n"));
+
+        const renameProvider: vscode.RenameProvider = {
+            provideRenameEdits(document: vscode.TextDocument, position: vscode.Position, newName: string, token: vscode.CancellationToken): vscode.WorkspaceEdit {
+                return self.onProvideRename(document, position, newName, token);
+            }
+        };
+        deploymentTemplateFileSubscriptionsArray.push(vscode.languages.registerRenameProvider(supportedDocumentSelector, renameProvider));
+
+        this._diagnosticsCollection = vscode.languages.createDiagnosticCollection("azurermtle");
+        deploymentTemplateFileSubscriptionsArray.push(this._diagnosticsCollection);
+
+        this._deploymentTemplateFileSubscriptions = vscode.Disposable.from(...deploymentTemplateFileSubscriptionsArray);
+
+        if (surveySettings.showSurveyPrompts) {
+            AzureRMAssets.getSurveyMetadata()
+                .then((surveyMetadata: SurveyMetadata) => {
+                    if (surveyMetadata.shouldShowSurveyPrompt(surveySettings)) {
+                        const message: string = "We would like to collect your feedback on the Azure Resource Manager Tools extension. Please take a few minutes to fill out survey.";
+                        const options: string[] = ["Don't ask me again", "OK"]; // vS Code automatically inserts a "Close" option.
```

Comment:
Comment - VSCode or vscode. vS Code looks off. I realize this is not code you changed. :)